### PR TITLE
Added travis cross compilation support for linux/solaris/windows

### PR DESCRIPTION
**Pull request by zquestz**
Automates having Travis CI build binaries for Linux, Solaris and Windows. It can support any other architectures that golang supports as long as they don't use `import "C"`.

Unfortunately when using CGO, the headers and libraries required to cross compile are not available. The only function currently complaining is `procUsage`.

You will need to create a `GITHUB_TOKEN` with repo privileges and assign that as an ENV var in Travis-CI to get everything operational.

@derekcollison @wallyqs

### CODE REVIEW

**Comment by derekcollison** [comment]
So this currently does not work correct?

**Comment by zquestz** [comment]
It works for the platforms included. Just doesn't include all the platforms we want.

**Comment by zquestz** [comment]
Yeah you shouldn't need mousetrap.

File: .travis.yml [hunk 1]
```diff
@@ -15,3 +15,5 @@ script:
 - go test -i -race ./...
 - go test -v -race ./...
 - ./scripts/cov.sh TRAVIS
+after_script:
+- if [ "$TRAVIS_GO_VERSION" = "1.5" ] && [ "$BUILD_GOOS" = "linux" ] && [ "$TRAVIS_TAG" != "" ]; then ./scripts/cross_compile.sh; ghr --username derekcollison --token $GITHUB_TOKEN --replace $TRAVIS_TAG pkg/; fi
```
Comment (wallyqs) [inline .travis.yml]: btw I think this one here would have to be `ghr --username nats-io` so that the built artifacts are put under `nats-io/gnatsd`

**Comment by mcqueary** [comment]
What's the next step on this, folks?

**Comment by zquestz** [comment]
This is ready to go for windows, solaris and linux. The travis file will push built binaries whenever you push a tag. 

Here's what the output looks like:
```
cross_compile ✔ $ ./scripts/cross_compile.sh       
Number of parallel builds: 3

-->     windows/386: github.com/zquestz/gnatsd
-->   windows/amd64: github.com/zquestz/gnatsd
-->       linux/386: github.com/zquestz/gnatsd
-->       linux/arm: github.com/zquestz/gnatsd
-->     linux/amd64: github.com/zquestz/gnatsd
-->   solaris/amd64: github.com/zquestz/gnatsd
make pkg/gnatsd-linux_386.zip
make pkg/gnatsd-linux_amd64.zip
make pkg/gnatsd-linux_arm.zip
make pkg/gnatsd-solaris_amd64.zip
make pkg/gnatsd-windows_386.zip
make pkg/gnatsd-windows_amd64.zip
```

However darwin doesn't function due to the following error:
```
1 errors occurred:
--> darwin/amd64 error: exit status 2
Stderr: # github.com/nats-io/gnatsd/server
../../nats-io/gnatsd/server/monitor.go:460: undefined: procUsage
```

Appears the `procUsage` function in the `pse_*.go` files break on some platforms.

**Comment by derekcollison** [comment]
Thanks, how do we fix for macosx since we do ship release builds for it?

**Comment by zquestz** [comment]
We figure out another way to refactor `procUsage` to avoid C functions not available on linux. Ideally we figure out a good way to implement `procUsage` in go to make it more portable.

File: .travis.yml [hunk 2]
```diff
@@ -15,3 +15,5 @@ script:
 - go test -i -race ./...
 - go test -v -race ./...
 - ./scripts/cov.sh TRAVIS
+after_script:
```
Comment (derekcollison) [inline .travis.yml]: Do I have to do this on travis? Can I run the build script locally?

File: scripts/cross_compile.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+go get github.com/inconshreveable/mousetrap
+go get github.com/mitchellh/gox
+go get github.com/tcnksm/ghr
+
+export APPNAME="gnatsd"
+export OSARCH="linux/386 linux/amd64 linux/arm solaris/amd64 windows/386 windows/amd64"
```
Comment (derekcollison) [inline scripts/cross_compile.sh]: If I fix mac, what do I add? darwin/amd64?

Comment (zquestz) [inline scripts/cross_compile.sh]: Yep =)

File: .travis.yml (hunk 2 above)
Comment (zquestz) [inline .travis.yml]: You can run the script locally for testing. =)

**Pull request merged**